src/decisionengine/framework/dataspace/datablock.py

```diff
@@ -382,6 +384,9 @@ def __getitem__(self, key, default=None):
         except KeyError:
             self.logger.error(f"Did not get key '{key}' in datablock __getitem__")
             value = default
+        except NoResultFound:
+            self.logger.error(f"Did not get key '{key}' in datablock __getitem__")
+            value = default
```

Comment:
The datablock is intended to be independent of the dataspace implementation.  It would therefore be better to convert any SQLAlchemy errors to a backend-free error type (e.g. `KeyError`) at the implementation level, inside of  `sqlalchemy_ds/datasource_api.py`.

Reply:
Took a stab at it and pushed up a possible implementation.